### ISSUE TITLE
p2p/e2e: Validators not connecting to each other

### ISSUE THREAD

**Issue by thanethomson** [cometbft/cometbft#576]
A follow-up to #575.

It appears as though, in our E2E tests on `v0.34.x`, when using the `test/e2e/networks/ci.toml` manifest, at least one of the validators (e.g. `validator02`) fails to connect to the other validators on the same network.

The full output log of such a run is available here: https://gist.github.com/thanethomson/3b85c8fa8ce3a731d748e52d5bdd0105

This was not reproducible on `v0.37.x` or `main`. It could be architecture- or platform-specific, as we have not yet been able to replicate this on macOS, and has only manifested on Ubuntu so far using Docker.

**Comment by thanethomson** [comment]
After applying the patch in #577 to `v0.34.x`, when encountering this issue, the E2E tests will hang indefinitely.

**Comment by cason** [comment]
So, from the logs, `validator02` connects to `seed02` at startup. It request peer addresses to the seed node, which does not report any new address (likely reports its own address), then disconnects from `validator02`, which is standard for a seed node. Since `seed02` is the only node `validator02` knows, it attempts connecting to it again. Since `validator02` is the only node `seed02` knows, it attempts connecting to it again, and succeeds.

`seed02` fails connecting to `seed01`, which the configured seed node, and vice-versa: `seed01` fails connecting to `seed02`. I couldn't understand why this happens. There are connections rejected because they are duplicated and EOF errors. It could be the case that both, by having two connections with the remote peer, one on each direction, choose the "wrong one" to close. The result is that no connection is established.

So, in summary, `validator02` and `seed02` are only able connect to each other, forming a network partition.

**Comment by cason** [comment]
Is this still an issue?

**Comment by sergio-mena** [comment]
I haven't seen it for ages, so if nobody saw it recently, I'd say we can close it for the time being (we can always reopen if needed)
